azplugins/test-py/test_bond_double_well.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2018-2020, Michael P. Howard
+# This file is part of the azplugins project, released under the Modified BSD License.
+
+# Maintainer: astatt
+
+from hoomd import *
+from hoomd import md
+context.initialize()
+try:
+    from hoomd import azplugins
+except ImportError:
+    import azplugins
+import unittest
+
+# azplugins.bond.double_well
+class bond_double_well_tests(unittest.TestCase):
+    def setUp(self):
+        snap = data.make_snapshot(N=2, box=data.boxdim(L=20), particle_types=['A'], bond_types=['bond'])
+
+        if comm.get_rank() == 0:
+            snap.bonds.resize(1)
+            snap.bonds.group[0] = [0,1]
+
+        self.s = init.read_snapshot(snap)
+        self.nl = md.nlist.cell()
+        context.current.sorter.set_params(grid=8)
+
+    # basic test of creation
+    def test(self):
+        double_well = azplugins.bond.double_well()
+        double_well.bond_coeff.set('bond', V_max=1.0, a=1.0, b=0.7)
+        double_well.update_coeffs()
+
+    # test missing coefficients
+    def test_set_missing_V_max(self):
+        double_well = azplugins.bond.double_well()
+        double_well.bond_coeff.set('bond',  a=3, b=1.5)
+        self.assertRaises(RuntimeError, double_well.update_coeffs)
+
+    def test_set_missing_a(self):
+        double_well = azplugins.bond.double_well()
+        double_well.bond_coeff.set('bond', V_max=1.0, b=1.5)
+        self.assertRaises(RuntimeError, double_well.update_coeffs)
+
+    def test_set_missing_b(self):
+        double_well = azplugins.bond.double_well()
+        double_well.bond_coeff.set('bond',  V_max=1.0, a=1.5)
+        self.assertRaises(RuntimeError, double_well.update_coeffs)
+
+    def test_set_zero_b(self):
+        double_well = azplugins.bond.double_well()
+        double_well.bond_coeff.set('bond',  V_max=1.0, a=1.5,b=0)
+        self.assertRaises(ValueError, double_well.update_coeffs)
+
+    def tearDown(self):
+        del self.s
+        context.initialize()
+
+# azplugins.bond.double_well
+class potential_bond_double_well_tests(unittest.TestCase):
+    def setUp(self):
+        snap = data.make_snapshot(N=2, box=data.boxdim(L=20), particle_types=['A'],bond_types = ['bond'])
+
+        if comm.get_rank() == 0:
+            snap.bonds.resize(1)
+            snap.bonds.group[0] = [0,1]
+            snap.particles.position[0] = (0,0,0)
+            snap.particles.position[1] = (1,0,0)
+
+        self.s = init.read_snapshot(snap)
+        self.nl = md.nlist.cell()
+        context.current.sorter.set_params(grid=8)
+
+    # test the calculation of force and potential
+    def test_potential_minimum(self):
+
```

Comment:
Remove extra whitespace from the beginning of this test and the ones that follow.